utils/featureFlags.ts

```diff
@@ -0,0 +1,33 @@
+import { right } from "fp-ts/lib/Either";
+import { fromEither, TaskEither } from "fp-ts/lib/TaskEither";
+import { InstallationId } from "../generated/notifications/InstallationId";
+import { NHPartitionFeatureFlag } from "./config";
+
+/**
+ *
+ * @param enabledFeatureFlag The feature flag currntly enabled
+ * @param sha the installation id of the user
+ * @returns `true` if the user is enabled for the new feature, `false` otherwise
+ */
+export const getIsInActiveSubset = (
+  isUserATestUser: (InstallationId) => TaskEither<Error, boolean>
+) => (
+  enabledFeatureFlag: NHPartitionFeatureFlag,
+  sha: InstallationId
+): TaskEither<Error, boolean> => {
+  switch (enabledFeatureFlag) {
+    case NHPartitionFeatureFlag.all:
+      return fromEither(right(true));
+
+    case NHPartitionFeatureFlag.beta:
+      return isUserATestUser(sha);
+
+    case NHPartitionFeatureFlag.canary:
+      // Todo
+      return fromEither(right(false));
+
+    case NHPartitionFeatureFlag.none:
+    default:
```

Comment:
Why a default case? I think we better just cover all possible cases. If we turn `NHPartitionFeatureFlag` into a union, Typescript itself would handle it with exhaustive checks

Reply:
Do you mean like this?
<img width="665" alt="Schermata 2021-04-10 alle 12 35 21" src="https://user-images.githubusercontent.com/75862507/114266905-4f396500-99f9-11eb-90ff-00a8e26bb158.png">

It can be achieved also with enum, and it's actually a good point, thanks!

Reply:
In a scenario like this, when the switch returns at each case (which is a best practice) we don't need to add an explicit assertion, because Typescript itself would throw an error anyway if you missed one case (saying that there are branches without return.

Reply:
Ok, I think I'm missing something, because I've got no errors from TS when using literals and removing a case statement.

And actually I don't like to put strings in code if not really necessary, but I'm moving to literals in new commit

Reply:
just want to point out that keyof is preferred over union of literals:

https://github.com/gcanti/io-ts/tree/1.x#union-of-string-literals

Reply:
Done as `keyof` in [888e53a](https://github.com/pagopa/io-functions-pushnotifications/pull/17/commits/888e53a33e5c65b25e5e322287cd8407a9389747)

Reply:
oh, we already had a PR for this: https://github.com/pagopa/io-functions-pushnotifications/pull/20

Reply:
> I don't remember if we had a conversation about this being acceptable, do you @gquadrati?
No, we didn't, but I knew there was a PR waiting to be considered.. I was just waiting the right moment to do it.. I think the moment has arrived